presto-orc/src/main/java/com/facebook/presto/orc/cache/StorageOrcFileTailSource.java

```diff
@@ -114,7 +139,15 @@ public OrcFileTail getOrcFileTail(OrcDataSource orcDataSource, MetadataReader me
         Slice metadataSlice = completeFooterSlice.slice(0, metadataSize);
```

Comment:
metadataREads from (0, metadataSize), footerReads from (metadataSize, footerSize) and dwrfStripeCacheSlice reads from (0, dwrfStripeCacheSize).

The offset computation looks incorrect.

Reply:
Great catch! Fixed + added an assert in the unit test.